src/common/core/coretype.js

```diff
@@ -604,13 +604,19 @@ define([
         };
 
         this.setPointer = function (node, name, target) {
+            var parent;
             innerCore.setPointer(node, name, target);
+
             if (isInheritedChild(node)) {
-                this.setProperty(node, CONSTANTS.INHERITED_CHILD_HAS_OWN_RELATION_PROPERTY, true);
+                self.setProperty(node, CONSTANTS.INHERITED_CHILD_HAS_OWN_RELATION_PROPERTY, true);
+                parent = self.getParent(node);
```

Comment:
why this is in a separate command when we do not check anything for the parent?
and what if the parent is null? (probably it is enough to handle inside the function, but somewhere we should deal with it)